### ISSUE TITLE
Appsignal is not stopping correctly

### ISSUE THREAD

**Issue by janhalama** [appsignal/appsignal-nodejs#418]
Hi there,
Appsignal is not stopping correctly which is causing jest tests to hang. You can reproduce this issue in following repository https://github.com/janhalama/appsignal-test by running `yarn test` command.

This is screenshot of the running test:
![image](https://user-images.githubusercontent.com/5206165/127658519-5e693d1e-4dc6-4029-8b30-2f158796eb47.png)

I tried to find answer in existing issues, found this one related https://github.com/appsignal/appsignal-nodejs/issues/398 but answers did not help me much.

**Comment by ErfanEbrahimnia** [comment]
I'm experiencing the same issue. Did you find a solution @janhalama?

**Comment by janhalama** [comment]
No i did not find solution.

**Comment by tombruijn** [comment]
Hi there, do you want to use AppSignal in the test environment? If not, I recommend disabling AppSignal in the test env like described on this page: https://docs.appsignal.com/nodejs/configuration/#disable-appsignal-for-tests

Alternatively you could call the `stop` function on the AppSignal object in a global callback after the test suite is done?

**Comment by freaz** [comment]
Thanks for reaching back.

Skipping AppSignal is possible, but not desired as we want run e2e test including AppSignal.

And calling stop in global callback (Jest's afterAll) doesn't work either.

**Comment by ErfanEbrahimnia** [comment]
Basically the issue is that the `stop` method isn't doing anything when called

**Comment by tombruijn** [comment]
I had a quick look. I ran it with `jest --detectOpenHandles` and it told me the problem was in the minutely probe process that was still running.

If I add this config to the example linked above, it will not enable the minutely probes and work around this issue. The `appsignal.stop` line can also be removed then.

```diff
diff --git app.js app.js
index 037c00a..228ea8e 100644
--- app.js
+++ app.js
@@ -7,6 +7,7 @@ function startApp() {
     active: true,
     name: "express-test",
     apiKey: process.env.APPSIGNAL_PUSH_API_KEY,
+    enableMinutelyProbes: false
   });
```

I hope that works for everyone for now. We'll discuss this issue with the team to figure out the best way to shut down the minutely probes process on `Appsignal.stop` or when the Node process stops.

**Comment by unflxw** [comment]
Hi @janhalama @ErfanEbrahimnia @freaz! This issue should be fixed in version 2.3.2 of the `@appsignal/nodejs` package. Calling `appsignal.stop()` after your tests (i.e. in an `afterAll()` block, or in [Jest's `globalTeardown` config property](https://jestjs.io/docs/configuration#globalteardown-string)) should clear all pending minutely probes, avoiding the issue where the process keeps running after the tests are done.

Please let us know if you run into any further issues with the package.
